osu.Game/Rulesets/Edit/Checks/CheckZeroLengthObjects.cs

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System.Collections.Generic;
+using osu.Game.Rulesets.Edit.Checks.Components;
+using osu.Game.Rulesets.Objects;
+using osu.Game.Rulesets.Objects.Types;
+
+namespace osu.Game.Rulesets.Edit.Checks
+{
+    public class CheckZeroLengthObjects : ICheck
+    {
+        /// <summary>
+        /// The duration can be this low before being treated as having no length, in case of precision errors. Unit is milliseconds.
+        /// </summary>
+        private const double leniency = 0.5d;
+
+        public CheckMetadata Metadata { get; } = new CheckMetadata(CheckCategory.Compose, "Zero-length hitobjects");
```

Comment:
```suggestion
        public CheckMetadata Metadata { get; } = new CheckMetadata(CheckCategory.Compose, "Zero-length objects");
```

We don't use the term "hitobject" anywhere in messages exposed to the user, as far as I can see.

Reply:
There are a few other checks where this term is used, like "unsnapped hitobjects", "concurrent hitobjects", "offscreen hitobjects", etc. I'd say if we want to change that it'll probably be best handled in that separate pass you mentioned on discord, as it will affect more than just this PR.

Reply:
Can probably happen when we add localisation support for all this stuff, if not sooner.